riscv/sim.cc

```diff
@@ -406,10 +406,9 @@ void sim_t::set_rom()
 char* sim_t::addr_to_mem(reg_t paddr) {
   if (!paddr_ok(paddr))
     return NULL;
-  auto desc = bus.find_device(paddr);
+  auto desc = bus.find_device(paddr >> PGSHIFT << PGSHIFT, PGSIZE);
   if (auto mem = dynamic_cast<abstract_mem_t*>(desc.second))
```

Comment:
afaik `dynamic_cast` of a `nullptr` has an undefined behavior. 
maybe is more safe to do check before, e.g. 
```
if (desc.second && (auto mem = dynamic_cast<abstract_mem_t*>(desc.second)))

Reply:
Pretty sure this is actually well defined to return a `nullptr` of the new type, see e.g. https://en.cppreference.com/w/cpp/language/dynamic_cast

"target-type is [...] pointer to Base and the type of expression is pointer to Derived such that Base is a base class of Derived, the result is
- a null pointer value if expression is a null pointer value, or [...]"